File: backend/app/schemas/estoque.py
"""Schemas Pydantic do módulo de Estoque (Prioridade 3)."""
from __future__ import annotations

from decimal import Decimal
from typing import Literal, Optional

from pydantic import BaseModel, ConfigDict


class LocalOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    nome: str
    tipo: str
    ativo: bool


class SaldoOut(BaseModel):
    produto_id: int
    sku_base: str
    nome_produto: str
    local_id: int
    local_nome: str
    qtd_disponivel: Decimal
    qtd_reservada: Decimal
    custo_medio: Decimal
    valor_total: Decimal


class MovimentoIn(BaseModel):
    produto_id: int
    local_id: int
    tipo: Literal["entrada", "saida", "reserva", "liberacao"]
    qtd: Decimal
    custo_unitario: Optional[Decimal] = None
    referencia: Optional[str] = None
    observacao: Optional[str] = None


class SaldoSimplesOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    produto_id: int
    local_id: int
    qtd_disponivel: Decimal
    qtd_reservada: Decimal
    custo_medio: Decimal


class AlertaOut(BaseModel):
    produto_id: int
    sku_base: str
    nome: str
    disponivel_total: Decimal
    estoque_minimo: Decimal


class RankingItemOut(BaseModel):
    sku_base: Optional[str]
    unidades: Decimal
    liquido: Decimal


class RelatorioEstoqueOut(BaseModel):
    valor_total_estoque: Decimal
    total_alertas: int
    alertas: list[AlertaOut]
    ranking_skus: list[RankingItemOut]

File: frontend/src/types/estoque.ts
export interface Local {
  id: number;
  nome: string;
  tipo: string;
  ativo: boolean;
}

export interface Saldo {
  produto_id: number;
  sku_base: string;
  nome_produto: string;
  local_id: number;
  local_nome: string;
  qtd_disponivel: string;
  qtd_reservada: string;
  custo_medio: string;
  valor_total: string;
}

export interface Alerta {
  produto_id: number;
  sku_base: string;
  nome: string;
  disponivel_total: string;
  estoque_minimo: string;
}

export interface RankingItem {
  sku_base: string | null;
  unidades: string;
  liquido: string;
}

export interface RelatorioEstoque {
  valor_total_estoque: string;
  total_alertas: number;
  alertas: Alerta[];
  ranking_skus: RankingItem[];
}

export type TipoMovimento = "entrada" | "saida" | "reserva" | "liberacao";

export interface MovimentoIn {
  produto_id: number;
  local_id: number;
  tipo: TipoMovimento;
  qtd: string;
  custo_unitario?: string | null;
  referencia?: string | null;
}
